lib/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+
+set(ofx_SRCS
+        messages.cpp
+        ofx_utilities.cpp
+        file_preproc.cpp
+        context.cpp
+        ofx_preproc.cpp
+        ofx_container_generic.cpp
+        ofx_container_main.cpp
+        ofx_container_security.cpp
+        ofx_container_statement.cpp
+        ofx_container_account.cpp
+        ofx_container_transaction.cpp
+        ofx_container_position.cpp
+        ofx_containers_misc.cpp
+        ofx_request.cpp
+        ofx_request_accountinfo.cpp
+        ofx_request_statement.cpp
+        ofx_sgml.cpp
+        ofc_sgml.cpp
+        win32.cpp
+        )
+
+set(ofx_HEADERS
+        messages.hh
+        ofx_preproc.hh
+        file_preproc.hh
+        context.hh
+        ofx_sgml.hh
+        ofc_sgml.hh
+        ofx_aggregate.hh
+        ofx_error_msg.hh
+        ofx_containers.hh
+        ofx_request.hh
+        ofx_request_accountinfo.hh
+        ofx_request_statement.hh
+        ofx_utilities.hh
+        tree.hh
+        win32.hh
+        )
+
+add_library(ofx ${ofx_SRCS} ${ofx_HEADERS})
```

Comment:
Similar to John, I'm puzzled by the missing SHARED (or STATIC) option here. This means when building everything by default, you just get a static library but no shared one - which is different to the autotools system that would by default build the shared one and not the static one.

I had to read about BUILD_SHARED_LIBS in the cmake docs because I didn't know about this (relatively new?) option, and the rest of your CMake files here doesn't mention this. Hence it also doesn't show up as an option in any cmake GUI such as `ccmake` or similar. Is there a possibility where you could highlight that this option could be set? And also set teh default to BUILD_SHARED_LIBRARY=on, to keep the same default behaviour as in autotools?

Reply:
Resorting to `BUILD_SHARED_LIBS` instead of marking it explicitly as SHARED/STATIC is preferred because it allows wrappers like `vcpkg` to make that decision depending on the use case. But as you suggest, adding a user-facing option to enable/disable it explicitly is indeed a good practice. Fixed.